static/src/javascripts/projects/commercial/modules/dfp/create-slot.js

```diff
@@ -0,0 +1,144 @@
+// @flow
+import adSizes from 'commercial/modules/ad-sizes';
+
+const inlineDefinition = {
+    sizeMappings: {
+        mobile: [adSizes.outOfPage, adSizes.empty, adSizes.mpu, adSizes.fluid],
+        desktop: [
+            adSizes.outOfPage,
+            adSizes.empty,
+            adSizes.mpu,
+            adSizes.video,
+            adSizes.video2,
+            adSizes.fluid,
+        ],
+    },
+};
+
+const adSlotDefinitions = {
+    im: {
+        label: false,
+        refresh: false,
+        sizeMappings: {
+            mobile: [
+                adSizes.outOfPage,
+                adSizes.empty,
+                adSizes.inlineMerchandising,
+                adSizes.fluid,
+            ],
+        },
+    },
+    'high-merch': {
+        label: false,
+        refresh: false,
+        name: 'merchandising-high',
+        sizeMappings: {
+            mobile: [
+                adSizes.outOfPage,
+                adSizes.empty,
+                adSizes.merchandisingHigh,
+                adSizes.fluid,
+            ],
+        },
+    },
+    'high-merch-lucky': {
+        label: false,
+        refresh: false,
+        name: 'merchandising-high-lucky',
+        sizeMappings: {
+            mobile: [adSizes.outOfPage, adSizes.empty, adSizes.fluid],
+        },
+    },
+    'high-merch-paid': {
+        label: false,
+        refresh: false,
+        name: 'merchandising-high',
+        sizeMappings: {
+            mobile: [
+                adSizes.outOfPage,
+                adSizes.empty,
+                adSizes.merchandisingHighAdFeature,
+                adSizes.fluid,
+            ],
+        },
+    },
+    inline: inlineDefinition,
+    mostpop: inlineDefinition,
+    comments: inlineDefinition,
+    'top-above-nav': {
+        sizeMappings: {
+            mobile: [
+                adSizes.outOfPage,
+                adSizes.empty,
+                adSizes.mpu,
+                adSizes.fabric,
+                adSizes.fluid,
+            ],
+        },
+    },
+};
+
+const createAdSlotElement = (
+    name: string,
+    attrs: Object,
+    classes: Array<string>
+) => {
+    const adSlot: HTMLDivElement = document.createElement('div');
+    adSlot.id = `dfp-ad--${name}`;
+    adSlot.className = `js-ad-slot ad-slot ${classes.join(' ')}`;
+    adSlot.setAttribute('data-link-name', `ad slot ${name}`);
+    adSlot.setAttribute('data-name', name);
```

Comment:
Probably not for this PR, but: those 2 calls could extend `attrs` and would be set during the next 3 following lines :)